### DISCUSSION TITLE
What are the differences / advantages to D3DMetal?

### DISCUSSION THREAD

**Discussion by hackgrid** [3Shain/dxmt#15] [General]
Besides being opensource :-)

**Comment by K0bin** [comment]
D3DMetal shares a lot of code between their D3D12 and D3D11 implementation, including the synchronization. So DXMT could very well end up quite a bit faster because it relies on Metals built-in tracking to handle synchronization instead of grossly over-syncing like D3DMetal does (because the D3D12 barrier model can't really be implemented very well on top of Metal).

**Comment by billhollings** [comment]
On a related subtopic...

I'm really impressed with the `airconv` shader converter here. Quite a project!

I'm very curious about what defined the decision to roll your own shader converter, instead of leveraging Apple's [Metal Shader Converter](https://developer.apple.com/metal/shader-converter/) from their GPTK?

**Reply by 3Shain** [reply]
For perfect integration. Metal already has a bunch of unpleasant limitations to work around so I have 0 interest to deal with another proprietary piece, and luckily I can afford the price: to create a FOSS alternative.
